components/Admin/Admin.js

```diff
@@ -22,35 +24,67 @@ class Admin extends React.Component {
 
   handleSearch = (event) => {
     const searchTerm = event.currentTarget.value
-    let { locations, filteredLocations } = this.state
+    let { locations, matchingSearchLocations } = this.state
 
     if (searchTerm) {
-      filteredLocations = locations.filter(location =>
+      matchingSearchLocations = locations.filter(location =>
```

Comment:
You're right. But we should also avoid `let` and reassigning variables if we can. I know JS makes that hard, but it makes functions more safe to edit as they grow in size/scope.